### DISCUSSION TITLE
Apache TinkerPop and Gremlin?

### DISCUSSION THREAD

**Discussion by spmallette** [TuGraph-family/tugraph-db#341] [General]
Hello, I recently came across these impressive benchmark results at LDBC for TuGraph: https://ldbcouncil.org/benchmarks/snb-bi/ - that is great work!

I noted that TuGraph used Gremlin for that benchmark which was interesting to me and other members of the TinkerPop Community because while we were aware of TuGraph we didn't know that it supported Gremlin. I did search through your documentation and website and didn't see any mention of Gremlin as a language that is supported. 

With all that being said, I suppose my question is: Does TuGraph support Gremlin and, if so, is there any information you could share about its functionality?  Thank you!

**Comment by fanzhidongyzby** [comment]
@spmallette 
Thank you for your interest and questions! 

Let me briefly address your concerns: TuGraph is the unified brand name for Ant Group's graph computing products, which includes a graph database, graph computing engine, graph learning framework, and graph development platform, forming a complete stack of graph ecosystem products. 

The TuGraph product stack is also divided into a community edition and a commercial edition. On GitHub, you can find all the source code and documentation for the TuGraph community edition, while the commercial edition is primarily dedicated to Ant Group's internal business, partner companies, and performance benchmarking tasks. 

The TuGraph commercial edition currently supports Gremlin/GQL, but in the future, we aim to gradually embrace the ISO standard and align more closely with GQL. Therefore, in the community edition, you will see that it currently only supports GQL/Cypher. 

Of course, we are not opposed to the possibility of the community edition supporting Gremlin in the future, and we would warmly welcome developers who can provide complete support for the Gremlin language for TuGraph.

**Reply by spmallette** [reply]
thank you for the information. that certainly clarifies things. would you have any interest in seeing TuGraph listed on the TinkerPop website with the other providers who support Gremlin? 

https://tinkerpop.apache.org/providers.html
